tests/foreman/api/test_syncplan.py

```diff
@@ -493,43 +495,144 @@ def test_positive_repeatedly_add_remove(self):
+
     @run_only_on('sat')
     @tier2
     def test_positive_synchronize_product(self):
-        """Create a sync plan with one product and sync it.
+        """Create a sync plan with one product and sync it automatically.
 
-        @Assert: A sync plan is created with one product and product can be
-        synchronized.
+        @Assert: Product is synchronized successfully.
 
         @Feature: SyncPlan
         """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
+        ).create()
+        product = entities.Product(organization=self.org).create()
+        repo = entities.Repository(product=product).create()
+        # Verify product is not synced and doesn't have any content
+        self.validate_repo_content(
+            repo, ['erratum', 'package', 'package_group'], after_sync=False)
+        # Associate sync plan with product
+        sync_plan.add_products(data={'product_ids': [product.id]})
+        # Verify product was synced successfully
+        self.validate_repo_content(
+            repo, ['erratum', 'package', 'package_group'])
 
-    @stubbed()
     @run_only_on('sat')
     @tier2
     def test_positive_synchronize_products(self):
-        """Create a sync plan with two products and sync them.
+        """Create a sync plan with multiple products and sync them afterwards
+        automatically.
+
+        @Assert: Multiple Products are synchronized successfully.
+
+        @Feature: SyncPlan
+        """
+        sync_plan = entities.SyncPlan(
+            organization=self.org,
+            enabled=True,
+            sync_date=datetime.utcnow(),
+        ).create()
+        products = [
+            entities.Product(organization=self.org).create()
+            for _ in range(random.randint(3, 5))
+        ]
+        repos = [
+            entities.Repository(product=product).create()
+            for product in products
+            for _ in range(random.randint(2, 3))
+        ]
+        for repo in repos:
+            self.validate_repo_content(
+                repo,
+                ['erratum', 'package', 'package_group'],
+                after_sync=False,
+            )
+        sync_plan.add_products(data={
+            'product_ids': [product.id for product in products]})
+        for repo in repos:
+            self.validate_repo_content(
+                repo, ['erratum', 'package', 'package_group'])
+
+    @tier4
+    def test_positive_synchronize_product_for_future_date(self):
+        """Create a sync plan with sync date in future and synchronize one
+        product using that sync plan automatically.
 
-        @Assert: A sync plan is created with one product and products can be
-        synchronized.
+        @Assert: Product is synchronized successfully.
 
         @Feature: SyncPlan
         """
+        delay = 15 * 60  # delay for sync date in seconds
```

Comment:
Can you explain how this works? The sync_plan is created with current time - `datetime.utcnow()`